BE/dns.py

```diff
@@ -138,8 +139,9 @@ def buildResponse(d, ADDR, PORT):
 
         fail_reply.add_answer(RR(rname = '.'.join(domain), rtype = rtype, rclass = 1, rdata = A(FAILURE_IP))) if USE_FAILURE else 0
         return fail_reply.pack()
-    subs = domain[:-3]
-    uuid = subs[-1]
+    subs = domain_not_splited.removesuffix(host_domain).strip(". ")
+    subs_splited = subs.split(".")
+    uuid = subs_splited[-1]
```

Comment:
Compared to the previous behavior, in the case `vulnerable.com.uuidhere.dnsfookup.tld` this would fail.
Hence I suggest something like

```py
def get_uuid(subdomain, host_domain):
     return subdomain
         .strip(host_domain)
         .split('.')[-2]
```